cron/data/iterator.go

```diff
@@ -83,9 +84,14 @@ func (reader *csvIterator) Next() (RepoFormat, error) {
 	if reader.err != nil {
 		return reader.next, fmt.Errorf("reader has error: %w", reader.err)
 	}
-	// Sanity check valid GitHub URL.
-	if _, err := githubrepo.MakeGithubRepo(reader.next.Repo); err != nil {
-		return reader.next, fmt.Errorf("invalid GitHub URL: %w", err)
+
+	repoURI := reader.next.Repo
+
+	// validate gitlab or github url
+	if _, err := gitlabrepo.MakeGitlabRepo(repoURI); err != nil {
+		if _, err := githubrepo.MakeGithubRepo(repoURI); err != nil {
+			return reader.next, fmt.Errorf("invalid URL, neither github nor gitlab: %w", err)
+		}
```

Comment:
this makes a call to the gitlab API correct?

Can we at least switch the ordering around so the cron isn't making 1.2M gitlab requests each week?

Reply:
It shouldn't do that; the solution is a little ugly but in `gitlabrepo.isvalid` we exit early if the `repo.host` is `github.com`.